dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/DateUnitPeriodTypeParser.java

```diff
@@ -117,7 +117,7 @@ private DateInterval parseInternal(
               week,
               calendar,
               PeriodType.MAP_WEEK_TYPE.get(periodType.getName()),
-              new DateTimeUnit(year, 1, 1));
+              periodType.adjustToStartOfWeek(new DateTimeUnit(year, 1, 4), calendar));
```

Comment:
@abyot, correct if I'm wrong. But I tried the following unit test, in the test class `NepaliCalendarTest`.

```
    dateTimeUnit = new DateTimeUnit(2025, 1, 4, true);
    period = periodType.createPeriod(dateTimeUnit, calendar);
    startDate = new DateTime(2024, 12, 30, 0, 0);
    endDate = new DateTime(2025, 1, 5, 0, 0);

    assertEquals("2025W1", period.getIsoDate());
    assertEquals(startDate.toDate(), period.getStartDate());
    assertEquals(endDate.toDate(), period.getEndDate());
```
And I get assertion error:

```
Expected :2025W1
Actual   :2024W52
```

These websites say it should be 2025W1, assuming my test is right:
`https://www.epochconverter.com/weeks/2025`
`https://isoweeks.com/`

Is my test correct? What do you think?

Thanks!

Reply:
Thanks @maikelarabori. Yes, it helps. Our calendar implementations assume the dateTimeUnit to be on the same calendar as the param calendar. In your case we have dateTimeUnit in iso but calendar in nepali, this is unlikely to happen in implementation usecase.

I have made fix for the calendar implementation to distiguish between iso and non-iso dateTimeUnit.